Submitted By: Joe Ciccone <jciccone@gmail.com>
Date: 07-25-2010
Initial Package Version: 3.4.6
Origin: Idea originally developed by Ryan Oliver and Greg Schafer for
        the Pure LFS project.
Upstream Status: Not Applied - CLFS Specific
Description: This patch makes changes to the paths for pure64, eg 64bit libs
             in /usr/lib, 32bit in /usr/lib32.

diff -Naur gcc-4.5.0.orig/gcc/config/i386/t-linux64 gcc-4.5.0/gcc/config/i386/t-linux64
--- gcc-4.5.0.orig/gcc/config/i386/t-linux64	2009-04-21 15:03:23.000000000 -0400
+++ gcc-4.5.0/gcc/config/i386/t-linux64	2010-07-25 16:24:14.609791880 -0400
@@ -25,7 +25,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_OSDIRNAMES = ../lib ../lib32
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
diff -Naur gcc-4.5.0.orig/gcc/config/mips/t-linux64 gcc-4.5.0/gcc/config/mips/t-linux64
--- gcc-4.5.0.orig/gcc/config/mips/t-linux64	2009-04-21 15:03:23.000000000 -0400
+++ gcc-4.5.0/gcc/config/mips/t-linux64	2010-07-25 16:24:14.609791880 -0400
@@ -18,7 +18,7 @@
 
 MULTILIB_OPTIONS = mabi=n32/mabi=32/mabi=64
 MULTILIB_DIRNAMES = n32 32 64
-MULTILIB_OSDIRNAMES = ../lib32 ../lib ../lib64
+MULTILIB_OSDIRNAMES = ../lib64 ../lib32 ../lib
 
 EXTRA_MULTILIB_PARTS=crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o
 
diff -Naur gcc-4.5.0.orig/gcc/config/rs6000/t-linux64 gcc-4.5.0/gcc/config/rs6000/t-linux64
--- gcc-4.5.0.orig/gcc/config/rs6000/t-linux64	2009-04-21 15:03:23.000000000 -0400
+++ gcc-4.5.0/gcc/config/rs6000/t-linux64	2010-07-25 16:24:14.609791880 -0400
@@ -36,7 +36,7 @@
 MULTILIB_EXTRA_OPTS     = fPIC mstrict-align
 MULTILIB_EXCEPTIONS     = m64/msoft-float
 MULTILIB_EXCLUSIONS     = m64/!m32/msoft-float
-MULTILIB_OSDIRNAMES	= ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib) nof
+MULTILIB_OSDIRNAMES	= ../lib ../lib32 nof
 MULTILIB_MATCHES        = $(MULTILIB_MATCHES_FLOAT)
 
 softfp_wrap_start := '\#ifndef __powerpc64__'
diff -Naur gcc-4.5.0.orig/gcc/config/s390/t-linux64 gcc-4.5.0/gcc/config/s390/t-linux64
--- gcc-4.5.0.orig/gcc/config/s390/t-linux64	2009-10-07 04:26:47.000000000 -0400
+++ gcc-4.5.0/gcc/config/s390/t-linux64	2010-07-25 16:24:14.609791880 -0400
@@ -1,3 +1,3 @@
 MULTILIB_OPTIONS = m64/m31
 MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 ../lib
+MULTILIB_OSDIRNAMES = ../lib ../lib32
diff -Naur gcc-4.5.0.orig/gcc/config/sh/linux.h gcc-4.5.0/gcc/config/sh/linux.h
--- gcc-4.5.0.orig/gcc/config/sh/linux.h	2007-08-02 06:49:31.000000000 -0400
+++ gcc-4.5.0/gcc/config/sh/linux.h	2010-07-25 16:24:14.609791880 -0400
@@ -47,7 +47,7 @@
 
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.2"
 
 #undef SUBTARGET_LINK_EMUL_SUFFIX
 #define SUBTARGET_LINK_EMUL_SUFFIX "_linux"
diff -Naur gcc-4.5.0.orig/gcc/config/sparc/linux.h gcc-4.5.0/gcc/config/sparc/linux.h
--- gcc-4.5.0.orig/gcc/config/sparc/linux.h	2009-12-07 18:31:01.000000000 -0500
+++ gcc-4.5.0/gcc/config/sparc/linux.h	2010-07-25 16:24:14.609791880 -0400
@@ -86,7 +86,7 @@
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib32 %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!ibcs: \
diff -Naur gcc-4.5.0.orig/gcc/config/sparc/linux64.h gcc-4.5.0/gcc/config/sparc/linux64.h
--- gcc-4.5.0.orig/gcc/config/sparc/linux64.h	2009-12-07 18:31:01.000000000 -0500
+++ gcc-4.5.0/gcc/config/sparc/linux64.h	2010-07-25 16:24:14.609791880 -0400
@@ -109,8 +109,8 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
@@ -121,7 +121,7 @@
   { "link_arch_default", LINK_ARCH_DEFAULT_SPEC },	  \
   { "link_arch",	 LINK_ARCH_SPEC },
 
-#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/usr/lib %{shared:-shared} \
+#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/usr/lib32 %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
@@ -130,7 +130,7 @@
         %{static:-static}}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
@@ -211,7 +211,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
diff -Naur gcc-4.5.0.orig/gcc/config/sparc/t-linux64 gcc-4.5.0/gcc/config/sparc/t-linux64
--- gcc-4.5.0.orig/gcc/config/sparc/t-linux64	2009-04-21 15:03:23.000000000 -0400
+++ gcc-4.5.0/gcc/config/sparc/t-linux64	2010-07-25 16:24:14.609791880 -0400
@@ -19,7 +19,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 ../lib
+MULTILIB_OSDIRNAMES = ../lib ../lib32
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
